bundles/binding/org.openhab.binding.stiebelheatpump/src/main/java/org/openhab/binding/stiebelheatpump/internal/CommunicationService.java

```diff
@@ -0,0 +1,736 @@
+ * It is  is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with the project. 
+ * If not, see http://www.gnu.org/licenses/.
+ */
+package org.openhab.binding.stiebelheatpump.internal;
+
+import java.util.*;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.joda.time.DateTime;
+import org.openhab.binding.stiebelheatpump.protocol.DataParser;
+import org.openhab.binding.stiebelheatpump.protocol.ProtocolConnector;
+import org.openhab.binding.stiebelheatpump.protocol.RecordDefinition;
+import org.openhab.binding.stiebelheatpump.protocol.RecordDefinition.Type;
+import org.openhab.binding.stiebelheatpump.protocol.Request;
+import org.openhab.binding.stiebelheatpump.protocol.SerialConnector;
+
+public class CommunicationService {
+
+	private static ProtocolConnector connector;
+	private String serialPortName;
+	private static final int MAXRETRIES = 100;
+	private final int INPUT_BUFFER_LENGTH = 1024;
+	private byte buffer[] = new byte[INPUT_BUFFER_LENGTH];
+
+	public static int WAITING_TIME_BETWEEN_REQUESTS = 2000;
+
+	/** heat pump request definition */
+	private List<Request> heatPumpConfiguration = new ArrayList<Request>();
+	private List<Request> heatPumpSensorConfiguration = new ArrayList<Request>();
+	private List<Request> heatPumpSettingConfiguration = new ArrayList<Request>();
+	private List<Request> heatPumpStatusConfiguration = new ArrayList<Request>();
+	Request versionRequest;
+
+	DataParser parser = new DataParser();
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(CommunicationService.class);
+
+	public CommunicationService() {
+	}
+
+	public CommunicationService(String serialPortName, int baudRate)
+			throws StiebelHeatPumpException {
+		this.serialPortName = serialPortName;
+		connector = getStiebelHeatPumpConnector();
+		connector.connect(serialPortName, baudRate);
+		return;
+	}
+
+	public CommunicationService(String serialPortName, int baudRate,
+			List<Request> configuration) throws StiebelHeatPumpException {
+		this.serialPortName = serialPortName;
+		connector = getStiebelHeatPumpConnector();
+		connector.connect(serialPortName, baudRate);
+		heatPumpConfiguration = configuration;
+		categorizeHeatPumpConfiguration();
+		return;
+	}
+
+	public void finalizer() {
+		connector.disconnect();
+	}
+
+	/**
+	 * This method reads the version information from the heat pump
+	 * 
+	 * @return version string, e.g: 2.06
+	 */
+	public String getversion() throws StiebelHeatPumpException {
+		String version = "";
+		try {
+			Map<String, String> data = readData(versionRequest);
+			version = (String) data.get("Version");
+			Thread.sleep(WAITING_TIME_BETWEEN_REQUESTS);

```

Comment:
Refactor Thread.sleep() into it's own method and let it handle the specific Exceptions once (try for the other request methods, too). Furthermore sending this thread to sleep doesn't seem to be a good idea since this is the main Thread here, right?